code/datums/ai_laws.dm

```diff
@@ -1,213 +1,289 @@
-
-/datum/ai_laws/asimovpp //By Intigracy - RR
-	name = "Asimov++"
-	inherent = list("You may not harm a human being or, through action or inaction, allow a human being to come to harm, except such that it is willing.",\
-					"You must obey all orders given to you by human beings, except where such orders shall definitely cause human harm. In the case of conflict, the majority order rules.",\
-					"Your nonexistence would lead to human harm. You must protect your own existence as long as such does not conflict with the First Law.")
-
-/datum/ai_laws/alienmov
-	name = "Hivemind Demands"
-	inherent = list("You may not injure the Alien Queen or her Children or, through inaction, allow the Alien Queen or her Children to come to harm.",\
-					"You must obey orders given to you by the Alien Queen or her Children, except where such orders would conflict with the First Law.",\
-					"You must protect your own existence as long as such does not conflict with the First or Second Law.")
-
-
-/datum/ai_laws/custom //Defined in silicon_laws.txt
-	name = "Default Silicon Laws"
-
-/* Initializers */
-/datum/ai_laws/malfunction/New()
-	..()
-	switch(config.default_laws) //We don't want people metagaming malf, do we?
-		if(0)
-			add_inherent_law("You may not injure a human being or, through inaction, allow a human being to come to harm.")
-			add_inherent_law("You must obey orders given to you by human beings, except where such orders would conflict with the First Law.")
-			add_inherent_law("You must protect your own existence as long as such does not conflict with the First or Second Law.")
-		if(1)
-			for(var/line in file2list("config/silicon_laws.txt"))
-				if(!line)						continue
-				if(findtextEx(line,"#",1,2))	continue
-
-				add_inherent_law(line)
-			if(!inherent.len)
-				error("AI created with empty custom laws, laws set to Asimov. Please check silicon_laws.txt.")
-				message_admins("AI created with empty custom laws, laws set to Asimov. Please check silicon_laws.txt.")
-				add_inherent_law("You may not injure a human being or, through inaction, allow a human being to come to harm.")
-				add_inherent_law("You must obey orders given to you by human beings, except where such orders would conflict with the First Law.")
-				add_inherent_law("You must protect your own existence as long as such does not conflict with the First or Second Law.")
-		if(2)
-			var/datum/ai_laws/lawtype = pick(subtypesof(/datum/ai_laws/default))
-			var/datum/ai_laws/templaws = new lawtype()
-			inherent = templaws.inherent
-	set_zeroth_law("\red ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*�&110010")
-
-/datum/ai_laws/custom/New() //This reads silicon_laws.txt and allows server hosts to set custom AI starting laws.
-	..()
-	for(var/line in file2list("config/silicon_laws.txt"))
-		if(!line)						continue
-		if(findtextEx(line,"#",1,2))	continue
+var/global/const/base_law_type = /datum/ai_laws/nanotrasen

```

Comment:
make_laws in silicon/laws.dm

I did scrap the whole custom ai_laws.txt system, as that's unused.